packages/front-end/components/AutoGenerateMetricsModal.tsx

```diff
@@ -166,8 +179,32 @@ export default function AutoGenerateMetricsModal({
   useEffect(() => {
     if (!selectedDatasource) return;
 
+    if (!selectedSchema && availableSchemas.length === 1) {
+      form.setValue("schema", availableSchemas[0]);
+    }
+
+    if (!selectedSchema) return;
+
     getTrackedEvents(selectedDatasource);
-  }, [getTrackedEvents, selectedDatasource]);
+  }, [
+    availableSchemas,
+    form,
+    getTrackedEvents,
+    selectedDatasource,
+    selectedSchema,
+  ]);
+
+  useEffect(() => {
+    if (selectedDatasourceData?.informationSchema) {
+      const schemas: string[] = [];
+      selectedDatasourceData.informationSchema.databases.forEach((database) => {
+        database.schemas.forEach((schema) => {
+          schemas.push(schema.schemaName);
+        });
+      });
+      setAvailableSchemas(schemas);
```

Comment:
I was able to get into a weird state where my information schema was stale and so my `availableSchemas` didn't include the one that actually had the segment `tracks` table. Should we consider:
(a) always fetching all available schemas if the information schema is stale (mine was over 6 weeks stale when I got in to this state
(b) at least update available schemas if we throw an error that says we can't find a relevant schema with metrics?

Reply:
Mmmm.... thats a tough one. 

Thankfully, in production, I don't think a user's info schema should ever be that stale - I think the job runs every 30 days, so it shouldn't get more than 4 weeks old.

However, 4 weeks is still quite a long time. I've added a "Refresh" button, like we have on the `SchemaBrowser` so a user can refresh the schema from the AutoGenerateMetricsModal. Let me know what you think of this treatment.

(There is a hover over the refresh button to provide some context)
<img width="616" alt="Screen Shot 2023-08-22 at 3 38 02 PM" src="https://github.com/growthbook/growthbook/assets/75274610/645d8269-fc35-4d7b-b33a-84e9f16c9438">

(The refresh kicks off an async job, so we have to poll the information schemas status to see when it's completed - this is a screenshot of the state when the info schema is mid-refresh)

<img width="518" alt="Screen Shot 2023-08-22 at 3 44 50 PM" src="https://github.com/growthbook/growthbook/assets/75274610/7a1f2272-c603-429f-a469-b182cba8bba8">

Reply:
The refresh is a good solution, but I'm not sure if it will solve the problem where we are throwing an error because the one data set we *do* have (if we only have one data set) doesn't have the `tracks` table or whatever.

Reply:
(edge case city, population: me)